api/v1alpha1/ingressnodefirewallnodestate_types.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// IngressNodeFirewallNodeStateSpec defines the desired state of IngressNodeFirewallNodeState
+type IngressNodeFirewallNodeStateSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// A list of ingress firewall policy rules.
+	// empty list indicates no ingress firewall i.e allow all incoming traffic.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Ingress []IngressNodeFirewallRules `json:"ingress"`
+
+	// A list of interfaces where the ingress firewall policy will be applied on.
+	// empty list indicates the firewall policy applied on all interfaces
+	// +kubebuilder:validation:Optional
+	// +optional
+	// +nullable
+	Interfaces *[]string `json:"interfaces"`
+}
```

Comment:
can u instead of redefine the same use the same type from fwrules object so we have one copy ?